src/ast/passes/codegen_llvm.cpp

```diff
@@ -1026,6 +1026,13 @@ void CodegenLLVM::visit(Call &call)
   {
     expr_ = b_.getInt64(call.vargs->at(0)->type.GetSize());
   }
+  else if (call.func == "offsetof")
+  {
+    auto &element_arg = *call.vargs->at(0);
+    String &element = static_cast<String&>(element_arg);
+    auto &field = call.vargs->at(1)->type.GetField(element.str);
```

Comment:
> What if the struct doesn't have the given field? This will throw an exception.

```
struct Foo {
    int x;
    long l;
    char c;
}
BEGIN
{
    printf("%ld\n", offsetof(y, struct Foo));
}
```

Will

```
ERROR: Failed to compile: struct has no field named y
```